docs/user-guide/expressions/lists.md

```diff
@@ -1,47 +1,102 @@
+
+This context is ideal for computing in row orientation.
+
+Polars expressions work on columns that have the guarantee that they consist of homogeneous data. Columns have this guarantee, rows in a `DataFrame` not so much. Luckily we have a data type that has the guarantee that the rows are homogeneous: `pl.List` data type.  We can **any** Polars operations on the elements of the list with the `list.eval` (`list().eval` in Rust) expression! These expressions run entirely on polars' query engine and can run in parallel so will be super fast.
 
-We can run **any** polars expression on the elements of a list with the `list.eval` (`list().eval` in Rust) expression! These expressions run entirely on polars' query engine and can run in parallel so will be super fast.
+Let's say we have another set of weather data across three days, for different stations:
+
+{{code_block('user-guide/expressions/lists','weather_by_day',['weather_by_day'])}}
+```python exec="on" result="text" session="user-guide/lists"
+--8<-- "python/user-guide/expressions/lists.py:weather_by_day"
+```
+
+Let's do something interesting. `pandas` allows you to compute the percentages of the rank values. `Polars` doesn't provide such a keyword argument. But because expressions are so versatile we can create our own percentage rank expression for highest temperature. Let's try that!
+
+{{code_block('user-guide/expressions/lists','weather_by_day_rank',['weather_by_day_rank'])}}
+```python exec="on" result="text" session="user-guide/lists"
+--8<-- "python/user-guide/expressions/lists.py:weather_by_day_rank"
+```
 
-Let's expand the example from above with something a little more interesting. Pandas allows you to compute the percentages of the `rank` values. Polars doesn't provide such a keyword argument. But because expressions are so versatile we can create our own percentage rank expression. Let's try that!
+# Polars Arrays
 
-Note that we must `select` the list's element from the context. When we apply expressions over list elements, we use `pl.element()` to select the element of a list.
+`Array`s are a new data type that was recently introduced, and are still pretty nascent in features that it offers. The major difference between a `List` and an `Array` is that the latter is limited to having the same number of elements per row, while a `List` can have a variable number of elements. Both still require that each element's data type is the same.
 
-{{code_block('user-guide/expressions/lists','expression',['list.eval'])}}
+We can define `Array` columns in this manner:
 
+{{code_block('user-guide/expressions/lists','array_df',['array_df'])}}
 ```python exec="on" result="text" session="user-guide/lists"
---8<-- "python/user-guide/expressions/lists.py:expression"
+--8<-- "python/user-guide/expressions/lists.py:array_df"
 ```
 
-Note that this solution works for any expressions/operation you want to do row wise.
+Basic operations are available on it (see Array for details).
```

Comment:
Could you provide a basic code snippet showing the functionality and then linking the user to the api documentation using the third argument and API_REFERENCE_LINKS